src/components/EventsList/EventsList.tsx

```diff
@@ -1,50 +1,53 @@
 import type { JSX } from 'solid-js/jsx-runtime';
 import styles from './styles.module.scss';
-import type { MeetupEventType } from '@api/meetup/event.graqhql.types';
-import { For } from 'solid-js';
+import { For, Show } from 'solid-js';
 import type { Lang } from '@i18n/types';
 import { CFPCta } from '@components/CFPCta/CFPCta';
 import { formatDate } from '@i18n/date-time';
+import type { UpcomingEvent } from 'utils/meetup-events';
 
 export interface EventsListProps {
-  data?: MeetupEventType[] | null;
-  upcoming?: Date[];
+  events?: UpcomingEvent[];
   lang: Lang;
 }
 
-export function EventsList({
-  data,
-  upcoming,
-  lang,
-}: EventsListProps): JSX.Element {
+export function EventsList({ events, lang }: EventsListProps): JSX.Element {
   return (
     <ul class={styles.eventsList}>
-      <For each={data}>
+      <For each={events}>
         {(event) => (
-          <li>
-            <section class={styles.event}>
-              <time datetime={new Date(event.dateTime).toISOString()}>
-                {formatDate(lang, new Date(event.dateTime))}
-              </time>
-              <h3 class={styles.eventHeading}>
-                <a href={event.eventUrl}>{event.title}</a>
-              </h3>
-            </section>
-          </li>
-        )}
-      </For>
-      <For each={upcoming}>
-        {(event) => (
-          <li>
-            <section class={styles.event}>
-              <time datetime={new Date(event).toISOString()}>
-                {formatDate(lang, new Date(event))}
-              </time>
-              <h3 class={styles.eventHeading}>
-                TBD <CFPCta />
-              </h3>
-            </section>
-          </li>
+          <>
+            <Show when={event.type === 'placeholder' && event.date}>
+              {(placeholderDate) => (
+                <li>
+                  <section class={styles.event}>
+                    <time datetime={placeholderDate().toISOString()}>
+                      {formatDate(lang, placeholderDate())}
+                    </time>
+                    <h3 class={styles.eventHeading}>
+                      TBD <CFPCta />
+                    </h3>
+                  </section>
+                </li>
+              )}
+            </Show>
+            <Show when={event.type === 'scheduled' && event.data}>
+              {(scheduled) => (
+                <li>
+                  <section class={styles.event}>
+                    <time
+                      datetime={new Date(scheduled().dateTime).toISOString()}
+                    >
+                      {formatDate(lang, new Date(scheduled().dateTime))}
+                    </time>
+                    <h3 class={styles.eventHeading}>
+                      <a href={scheduled().eventUrl}>{scheduled().title}</a>
+                    </h3>
+                  </section>
+                </li>
+              )}
+            </Show>
+          </>
```

Comment:
This component is becoming a bit hard to read.

How do you feel about splitting in multiple components?

Reply:
done in [feb1a5c](https://github.com/Roma-JS/roma-js-on-astro/pull/71/commits/feb1a5cb43c729605044f7f21757f7689dbf1c09)